### PR TITLE
Bump git-init version to v0.17.3

### PR DESCRIPTION
**Pull request by devholic**
# Changes

- Bump git-init version to v0.17.3 which contains bug-fixes for `git-init` ([Release Note](https://github.com/tektoncd/pipeline/releases/tag/v0.17.3))
  - Also, make git-init image in git-clone/clone as configurable value

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

### CODE REVIEW

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test
/lgtm

File: task/git-clone/0.2/git-clone.yaml [hunk 1]
```diff
@@ -68,14 +68,18 @@ spec:
       description: log the commands used during execution
       type: string
       default: "true"
+    - name: gitInitImage
```
Comment (Divyansh42) [inline task/git-clone/0.2/git-clone.yaml:71]: We will need to add this param to the README of the task as well. 
Can you please add that [here](https://github.com/tektoncd/catalog/blob/master/task/git-batch-merge/0.2/README.md)

Comment (Divyansh42) [reply task/git-clone/0.2/git-clone.yaml:71]: And Probably in other PR we will have to modify/cleanup README of this task like we have for other git [tasks](https://github.com/tektoncd/catalog/tree/master/task/git-cli/0.1)

Comment (devholic) [reply task/git-clone/0.2/git-clone.yaml:71]: Sure 🙂

**Comment by Divyansh42** [comment]
@devholic Can you please squash the commits as well?

**Comment by devholic** [comment]
I squashed commits 🙂

**Review by Divyansh42 — APPROVED** [review]
Looks good, Thanks!

/lgtm

**Review by vdemeester — APPROVED** [review]

**Pull request merged**